src/components/editor-page/editor-pane/linter/single-line-regex-linter.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import type { Linter } from './linter'
+import type { EditorView } from '@codemirror/view'
+import type { Diagnostic } from '@codemirror/lint'
+import { t } from 'i18next'
+
+/**
+ * Creates a {@link Linter linter} from {@link RegExp regexps} for single lines.
+ *
+ * @param regex The {@link RegExp regexp} to execute on each line
+ * @param message The message to display if the {@link RegExp regexp} hits
+ * @param replace The function to replace what was found by {@link RegExp regexp}
+ * @param actionLabel The optional label to translate and use as the fix button for the linter.
+ */
+export class SingleLineRegexLinter implements Linter {
+  constructor(
+    private regex: RegExp,
+    private message: string,
+    private replace: (match: string) => string,
+    private actionLabel?: string
+  ) {}
+
+  lint(view: EditorView): Diagnostic[] {
+    const lines = view.state.doc.toString().split('\n')
+    const diagnostics: Diagnostic[] = []
+    let from = 0
+    lines.forEach((line) => {
```

Comment:
I would prefer a reduce here. But that is personal preference and only an optional suggestion.